### PR TITLE
Fix the broken video orientation (+-90/180)

### LINKED ISSUE #6305: Videos Recorded in Portrait Mode are Rotated -90 Degrees When Transcoded using (h264_nvenc)
**Describe the bug**

Videos Recorded in Portrait Mode are Rotated -90 Degrees When Transcoded using (h264_nvenc). Believe related to the "displaymatrix: rotation of -90.00 degrees" parameter added to the end of the nvenc ffmpeg command.

**System (please complete the following information):**
 - OS: [Windows]
 - Clients: [Windows, ios]
 - Browser: [Firefox 72]
 - Jellyfin Windows Server Version [10.7.6]
 - Playback: [Transcode NVENC] 
 - Reverse Proxy: [nginx]

**To Reproduce**

1. Enable NVENC 264 transcoding
2. play video that has been recorded in portait mode using transcoding and nvenc.
3. result is video played back at a -90 degree rotaton.
 
**Expected behavior**

The video should play back in the same orientation as the non trans-coded video plays. 

**FFmpeg Logs**
------------------------------------------------------
**LIBX264**

`  Duration: 00:01:42.85, start: 0.000000, bitrate: 15528 kb/s
    Stream #0:0(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p(tv, bt709), 1920x1080, 15371 kb/s, 30 fps, 30 tbr, 600 tbn, 1200 tbc (default)
    Metadata:
      rotate          : 90
      creation_time   : 2021-05-21T07:02:15.000000Z
      handler_name    : Core Media Video
      encoder         : H.264
    Side data:
      displaymatrix: rotation of -90.00 degrees
    Stream #0:1(und): Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, mono, fltp, 100 kb/s (default)
    Metadata:
      creation_time   : 2021-05-21T07:02:15.000000Z
      handler_name    : Core Media Audio
    Stream #0:2(und): Data: none (mebx / 0x7862656D) (default)
    Metadata:
      creation_time   : 2021-05-21T07:02:15.000000Z
      handler_name    : Core Media Metadata
    Stream #0:3(und): Data: none (mebx / 0x7862656D), 9 kb/s (default)
    Metadata:
      creation_time   : 2021-05-21T07:02:15.000000Z
      handler_name    : Core Media Metadata
    Stream #0:4(und): Data: none (mebx / 0x7862656D), 37 kb/s (default)
    Metadata:
      creation_time   : 2021-05-21T07:02:15.000000Z
      handler_name    : Core Media Metadata
Stream mapping:
  Stream #0:0 -> #0:0 (h264 (native) -> h264 (libx264))
  Stream #0:1 -> #0:1 (copy)
Press [q] to stop, [?] for help
[libx264 @ 00000194e0edd280] using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2 AVX FMA3 BMI2 AVX2
[libx264 @ 00000194e0edd280] profile Main, level 4.1, 4:2:0, 8-bit
[libx264 @ 00000194e0edd280] 264 - core 159 r2991 1771b55 - H.264/MPEG-4 AVC codec - Copyleft 2003-2019 - http://www.videolan.org/x264.html - options: cabac=1 ref=1 deblock=1:0:0 analyse=0x1:0 me=dia subme=0 psy=1 psy_rd=1.00:0.00 mixed_ref=0 me_range=4 chroma_me=0 trellis=0 8x8dct=0 cqm=0 deadzone=21,11 fast_pskip=1 chroma_qp_offset=0 threads=24 lookahead_threads=4 sliced_threads=0 nr=0 decimate=1 interlaced=0 bluray_compat=0 constrained_intra=0 bframes=3 b_pyramid=2 b_adapt=1 b_bias=0 direct=1 weightb=1 open_gop=0 weightp=1 keyint=250 keyint_min=25 scenecut=40 intra_refresh=0 rc_lookahead=10 rc=crf mbtree=1 crf=23.0 qcomp=0.60 qpmin=0 qpmax=69 qpstep=4 vbv_maxrate=319 vbv_bufsize=639 crf_max=0.0 nal_hrd=none filler=0 ip_ratio=1.40 aq=1:1.00
Output #0, hls, to 'C:\ProgramData\Jellyfin\Server\transcodes\9f2947bad90ef32d672e1e7b3198b2c9.m3u8':
  Metadata:
    encoder         : Lavf58.45.100
    Stream #0:0: Video: h264 (libx264), yuv420p, 426x756, q=-1--1, 30 fps, 90k tbn, 30 tbc (default)
    Metadata:
      encoder         : Lavc58.91.100 libx264
    Side data:
      cpb: bitrate max/min/avg: 319000/0/0 buffer size: 639000 vbv_delay: N/A
      displaymatrix: rotation of -0.00 degrees`

RESULT:

firefox_Su2KT9Hn49

------------------------------------------------------
**H264_NVENC**

`  Duration: 00:01:42.85, start: 0.000000, bitrate: 15528 kb/s
    Stream #0:0(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p(tv, bt709), 1920x1080, 15371 kb/s, 30 fps, 30 tbr, 600 tbn, 1200 tbc (default)
    Metadata:
      rotate          : 90
      creation_time   : 2021-05-21T07:02:15.000000Z
      handler_name    : Core Media Video
      encoder         : H.264
    Side data:
      displaymatrix: rotation of -90.00 degrees
    Stream #0:1(und): Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, mono, fltp, 100 kb/s (default)
    Metadata:
      creation_time   : 2021-05-21T07:02:15.000000Z
      handler_name    : Core Media Audio
    Stream #0:2(und): Data: none (mebx / 0x7862656D) (default)
    Metadata:
      creation_time   : 2021-05-21T07:02:15.000000Z
      handler_name    : Core Media Metadata
    Stream #0:3(und): Data: none (mebx / 0x7862656D), 9 kb/s (default)
    Metadata:
      creation_time   : 2021-05-21T07:02:15.000000Z
      handler_name    : Core Media Metadata
    Stream #0:4(und): Data: none (mebx / 0x7862656D), 37 kb/s (default)
    Metadata:
      creation_time   : 2021-05-21T07:02:15.000000Z
      handler_name    : Core Media Metadata
Stream mapping:
  Stream #0:0 -> #0:0 (h264 (native) -> h264 (h264_nvenc))
  Stream #0:1 -> #0:1 (copy)
Press [q] to stop, [?] for help
Output #0, hls, to 'C:\ProgramData\Jellyfin\Server\transcodes\470b56c8671b84c28231aed288ad8cc2.m3u8':
  Metadata:
    encoder         : Lavf58.45.100
    Stream #0:0: Video: h264 (h264_nvenc) (High), cuda, 426x238, q=-1--1, 319 kb/s, 30 fps, 90k tbn, 30 tbc (default)
    Metadata:
      encoder         : Lavc58.91.100 h264_nvenc
    Side data:
      cpb: bitrate max/min/avg: 319892/0/319892 buffer size: 639784 vbv_delay: N/A
      displaymatrix: rotation of -90.00 degrees`

RESULT:

firefox_Srh4floiMA

**Additional context**

Auto - Direct play will play back the video without the -90 Rotation, as does when trans-coded with the CPU libx264, it's only when run through NVENC being trans-coded and videos taken in "portrait" that the -90 degree rotation occurs. There is no issue with videos originally recorded in "landscape" mode

### LINKED ISSUE #11102: [Issue]: HEVC transcoding flips video 180
### Please describe your bug

Installed [installer/jellyfin_10.8.13_windows-x64.exe](https://repo.jellyfin.org/releases/server/windows/stable/installer/jellyfin_10.8.13_windows-x64.exe), running latest version of Roku app/firefox/chrome.

On some GoPro footage I have (HEVC) the footage gets flipped 180 during transcoding, which appears on both Firefox and the Roku app (both presumably because they don't have that codec). Checking the logs, every video gets this 180 rotation added at the last line. I noticed the same when my phone records with HEVC, so not a GoPro issue.

```
  Stream #0:0[0x1](eng): Video: hevc (Main) (hvc1 / 0x31637668), yuvj420p(pc, bt709), 3840x2160 [SAR 1:1 DAR 16:9], 59927 kb/s, 23.98 fps, 23.98 tbr, 24k tbn (default)
    Metadata:
      creation_time   : 2023-06-25T13:36:50.000000Z
      handler_name    : GoPro H.265
      vendor_id       : [0][0][0][0]
      encoder         : GoPro H.265 encoder
      timecode        : 13:36:01:11
    Side data:
      displaymatrix: rotation of -180.00 degrees
```

Is there a configuration I can tweak somewhere so it won't rotate the file during transcoding? Using Intel QSV. I'm trying to avoid encoding the file to something else.

### Reproduction Steps

1. Setup Jellyfin with Intel QSV.
2. Load up HEVC from GoPro 10.
3. Stream to roku app or firefox.
4. Stream to Chrome or Edge to see it without transcoding.

### Jellyfin Version

10.8.13

### if other:

_No response_

### Environment

```markdown
- OS: Windows 11 Pro
- Linux Kernel: N/AA
- Virtualization: N/A
- Clients: Roku
- Browser: Firefox
- FFmpeg Version: Built into "installer" app (5.1.4-Jellyfin in C:\Program Files\Jellyfin\Server)
- Playback Method: Transcode
- Hardware Acceleration: Intel QSV
- GPU Model: Intel UHD Graphics (Alder Lake-N 041 GT1)
- Plugins:
- Reverse Proxy:
- Base URL:
- Networking:
- Storage:
```

### Jellyfin logs

```shell
N/A
```

### FFmpeg logs

```shell
See above.
```

### Please attach any browser or client logs here

_No response_

### Please attach any screenshots here

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by nyanmisaka**
Videos captured by mobile phones and action cameras such as GoPro often contain orientation information. The memory layout of such videos needs to be transposed so that they can be displayed on the playback device in the correct orientation.

In the past this was not supported in HWA because the corresponding hardware filter was not available. This problem was solved by writing the missing GPGPU based hardware filters.

**Changes**
- Add json parser for video rotation side data 
- Register video rotation side data in DB 
- Add tests for the video rotation side data 
- Only streamCopy videos containing rotation data in fMP4 
- Add HWA video transpose support (+-90/180) 

**Issues**
- Fixes #6305
- Fixes #11102
- https://forum.jellyfin.org/t-all-transcoded-videos-upside-down

### CODE REVIEW

File: MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs [hunk 1]
```diff
@@ -3016,8 +3023,10 @@ public MediaStream GetMediaStream(IEnumerable<MediaStream> allStream, int? desir
         }
 
         public static string GetHwScaleFilter(
+            string hwScalePrefix,
```
Comment (gnattu) [inline MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs:3026]: Is there any or will there be any scale filter does not use the name `scale`?

Comment (nyanmisaka) [reply MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs:3026]: It is unlikely to happen. When new hw scale filters are added, they need to follow the existing naming logic: `{scale,vpp}_{libname}`.

**Review by crobibero — APPROVED** [review]

**Review by Shadowghost — APPROVED** [review]

**Review by gnattu — APPROVED** [review]

File: MediaBrowser.Model/Entities/MediaStream.cs [hunk 2]
```diff
@@ -123,6 +123,12 @@ public class MediaStream
         /// <value>The Dolby Vision bl signal compatibility id.</value>
         public int? DvBlSignalCompatibilityId { get; set; }
 
+        /// <summary>
+        /// Gets or sets the Rotation.
```
Comment (Bond-009) [inline MediaBrowser.Model/Entities/MediaStream.cs:127]: Maybe add that it's # of degrees?

File: MediaBrowser.MediaEncoding/Probing/MediaStreamInfoSideData.cs [hunk 3]
```diff
@@ -69,5 +69,12 @@ public class MediaStreamInfoSideData
         /// <value>The DvBlSignalCompatibilityId.</value>
         [JsonPropertyName("dv_bl_signal_compatibility_id")]
         public int? DvBlSignalCompatibilityId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Rotation.
```
Comment (Bond-009) [inline MediaBrowser.MediaEncoding/Probing/MediaStreamInfoSideData.cs:74]: Same

**Review by Bond-009 — APPROVED** [review]

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**